src/sentry/testutils/hybrid_cloud.py

```diff
@@ -162,31 +153,32 @@ def simulate_on_commit(request: Any):
     from django.test import TestCase as DjangoTestCase
 
     request_node_cls = request.node.cls
+    is_django_test_case = request_node_cls is not None and issubclass(
+        request_node_cls, DjangoTestCase
+    )
     simulated_transaction_watermarks.state = {}
 
-    if request_node_cls is None or not issubclass(request_node_cls, DjangoTestCase):
-        yield
-        return
-
     _old_atomic_exit = transaction.Atomic.__exit__
     _old_transaction_on_commit = transaction.on_commit
 
     def maybe_flush_commit_hooks(connection):
-        if (
-            connection.in_atomic_block
-            and len(connection.savepoint_ids)
-            <= simulated_transaction_watermarks.state[connection.alias or "default"]
-            and not connection.closed_in_transaction
-            and not connection.needs_rollback
-        ):
-            old_validate = connection.validate_no_atomic_block
-            connection.validate_no_atomic_block = lambda: None
-            try:
-                connection.run_and_clear_commit_hooks()
-            finally:
-                connection.validate_no_atomic_block = old_validate
-        elif not connection.in_atomic_block or not connection.savepoint_ids:
-            assert not connection.run_on_commit, "Incidental run_on_commits detected!"
+        if connection.closed_in_transaction or connection.needs_rollback:
```

Comment:
This logic is really frustrating, but the goal is simple.  We're determining if, given the acceptable level of outer transactions, we should be flushing on_commit hooks.  Previously this logic would only work for `DjangoTestCase` but I needed it work for `DjangoTransactionalTestCase` as well to support better handling of SPLIT_DB mode.